x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/FrozenIndexInput.java

```diff
@@ -467,7 +467,10 @@ protected ByteRange maybeReadFromBlobCache(long position, int length) {
 
     private static int positionalWrite(SharedBytes.IO fc, long start, ByteBuffer byteBuffer) throws IOException {
         assert assertCurrentThreadMayWriteCacheFile();
-        return fc.write(byteBuffer, start);
+        byteBuffer.flip();
+        int written = fc.write(byteBuffer, start);
+        byteBuffer.clear();
```

Comment:
Can we clear buffer if not everything was written? Or would we have to assert that buffer.remaining() is 0 here and that written == old(buffer.remaining())?

Reply:
I think just  `assert byteBuffer.hasRemaining() == false;` is sufficient. We do assert that the sum of all writes matches up in the callers anyway and we don't have to add another variable just for the assertion here. I pushed https://github.com/elastic/elasticsearch/pull/70399/commits/34b179ea7cea42ad49f433b895d2bf20d9b409d2 for that.